src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -648,11 +670,12 @@ private void UpdateConsecutiveErrorCountOrThrow(IList<ChatMessage> added, ref in
             consecutiveErrorCount = 0;
         }
     }
+#pragma warning restore CA1851
 
     /// <summary>
     /// Throws an exception if <see cref="CreateResponseMessages"/> doesn't create any messages.
     /// </summary>
-    private void ThrowIfNoFunctionResultsAdded(IList<ChatMessage>? messages)
+    protected void ThrowIfNoFunctionResultsAdded(IList<ChatMessage>? messages)
```

Comment:
Same point as for `UpdateConsecutiveErrorCountOrThrow`

Reply:
This can be gone if we can configure the behavior of terminating early function invocation while looping.

Reply:
This doesn't need to be protected regardless. It's just a throw helper; there's no need for it to be exposed.